src/rattler_build_conda_compat/loader.py

```diff
@@ -94,12 +114,27 @@ def remove_empty_keys(variant_dict: dict[str, Any]) -> dict[str, Any]:
     return filtered_dict
 
 
+def flatten_lists(variant_dict: dict[str, Any]) -> dict[str, Any]:
```

Comment:
Are these methods considered to be used outside of this class? E.g. are they public? If not maybe we should prepend them with a `_`. Otherwise, disregard this comment.

Reply:
I've moved flatten_lists and remove_empty_keys in `utils.py` so it could be reused in future.
wdyt?

Reply:
`utils.py` is still a "public" module though. Personally I would either make `utils` "non-public" by prepending a `_` or prepending the functions.

The thing is that if we don't mark them as non-public they are public and people might start depending on these symbols.

Reply:
sorry but why it's a bad idea to expose this utills functions for the public?
flattening a lists and remove empty keys seems like a generic problem that isn't a rattler/conda specific

Reply:
I think it's a preference but to me, the scope of this library is _not_ providing utility functions to work with lists and dictionaries but rather to work with `recipe.yaml`s. In my eyes, we should be really careful to increase the API of the library. People _will_ start depending on anything we expose publicly which makes it harder for us to refactor things in the library later on. For instance, just moving a function from one module to another will then already be a breaking change.

WDYT @Hofer-Julian @tdejager

Reply:
got it, let me make them private

Reply:
I agree with @baszalmstra.

Regarding making all functions private vs making the module private, I think making the functions private is the way to go. It is more common in my experience.